scripts/build/builders/nuttx.py

```diff
@@ -0,0 +1,137 @@
+
+    def AppNamePrefix(self, chip_name):
+        if self == NuttXApp.LIGHT:
+            return ('chip-%s-lighting-example' % chip_name)
+        else:
+            raise Exception('Unknown app type: %r' % self)
+
+
+class NuttXBoard(Enum):
+    SIM = auto()
+
+
+def NuttXTarget(board, app):
+    if board == NuttXBoard.SIM:
+        if app == NuttXApp.LIGHT:
+            return 'sim:matter'
+
+
+class NuttXBuilder(Builder):
+
+    def __init__(self,
+                 root,
+                 runner,
+                 app: NuttXApp = NuttXApp.LIGHT,
+                 board: NuttXBoard = NuttXBoard.SIM,
+                 enable_rpcs: bool = False,
+                 module_type: str = 'sim',
+                 baudrate=2000000,
+                 enable_shell: bool = False,
+                 enable_cdc: bool = False,
+                 enable_resetCnt: bool = False,
+                 enable_rotating_device_id: bool = False,
+                 enable_ethernet: bool = False,
+                 enable_wifi: bool = False,
+                 enable_thread: bool = False,
+                 enable_frame_ptr: bool = False,
+                 enable_heap_monitoring: bool = False,
+                 use_matter_openthread: bool = False,
+                 ):
+
+        nuttx_chip = 'nuttx'
+
+        super(NuttXBuilder, self).__init__(
+            root=os.path.join(root, 'examples',
+                              app.ExampleName(), nuttx_chip),
+            runner=runner
+        )
+
+        self.argsOpt = []
+        self.chip_name = nuttx_chip
+        self.enable_frame_ptr = enable_frame_ptr
+        self.app = app
+        self.board = board
+        # self.nuttx_dir = os.path.join(self.output_dir, 'nuttx')
+        # self.nuttx_apps_dir = os.path.join(self.output_dir, 'apps')
+
+    def print_enviroment_error(self):
+        logging.fatal('*' * 80)
+        logging.fatal('*' * 80)
+
+    def generate(self):
+        logging.info('generate NuttX %s at %s',
+                     NuttXTarget(self.board, self.app), self.output_dir)
+
+        self._Execute(['mkdir', '-p', self.output_dir],
+                      title='Generating ' + self.identifier)
+
+        nuttx_dir = os.path.join(os.sep, 'opt', 'nuttx', 'nuttx')
+        self._Execute(['sed', '-i', 's/set(CMAKE_CXX_STANDARD 14)//g', os.path.join(nuttx_dir, 'CMakeLists.txt')])
+        config_path = os.path.join(nuttx_dir, 'boards', 'sim', 'sim', 'sim', 'configs', 'matter', 'defconfig')
+        self._Execute(['sed', '-i', 's/CONFIG_SIM_M32=y//g', config_path])
+
+        nuttx_apps_dir = os.path.join(os.sep, 'opt', 'nuttx', 'apps')
+        chip_root = os.getenv('PW_PROJECT_ROOT')
+        self._Execute(['ln', '-s', chip_root, os.path.join(nuttx_apps_dir, 'netutils', 'connectedhomeip', 'connectedhomeip')])
+
+        gcc13_bin_path = os.path.join(os.sep, 'opt', 'nuttx', 'gcc-13', 'bin')
+        self._Execute(['ln', '-s', os.path.join(gcc13_bin_path, 'gcc'), os.path.join(gcc13_bin_path, 'cc')])
```

Comment:
same comment with the ln here: these operations seem suspect as they are outside the build directory.

Reply:
Done.

Reply:
was the command altered to only change things in output_dir ? I seem to see the paths outside.